zio-http/src/main/scala/zio/http/codec/ContentCodecs.scala

```diff
@@ -24,6 +24,6 @@ private[codec] trait ContentCodecs {
   def content[A](implicit schema: Schema[A]): ContentCodec[A] =
     HttpCodec.Content(schema)
 
-  def contentStream[A](implicit schema: Schema[A]): ContentCodec[ZStream[Any, Nothing, A]] =
+  def contentStream[A](implicit schema: Schema[A]): ContentCodec[ZStream[Any, Throwable, A]] =
```

Comment:
Maybe this should not be changed and the input stream's errors should be handled somehow before passing the stream to the implementation, but I did not see an obvious place to do that.

Reply:
This should definitely not be changed.

The reason is that an endpoint handler cannot deal with `Throwable` errors. It only knows about typed errors, which are part of the protocol itself. `Throwable` errors indicate something fatal happened: for example, that an element of the stream was improperly encoded. No recovery is possible at the level of the endpoint handler. So it's a defect and must be handled as a fallback higher up (request handler, HTTP).